core/trino-main/src/main/java/io/trino/operator/project/PageProcessor.java

```diff
@@ -329,8 +329,9 @@ private ProcessBatchResult processBatch(int batchSize)
                 }
                 else {
                     if (pageProjectWork == null) {
+                        Page loadedPage = projection.getInputChannels().getInputChannels(page);
```

Comment:
where it will be accounted for?

Reply:
I'm not yet sure that this needs to be accounted for anywhere.
My thought was that if ExpressionProfiler is about detecting time consuming expression evaluation, then the addition of page loading time, which includes orc/parquet read from filesystem and decoding, should not be added to it.
But I wasn't certain about it.

Reply:
> I'm not yet sure that this needs to be accounted for anywhere.

i would bet that it needs to 

> My thought was that if ExpressionProfiler is about detecting time consuming expression evaluation

sounds so.

Reply:
cc @sopel39

Reply:
I did some benchmarking found no change to TPC and small improvement in scan operator benchmarks.
<img width="435" alt="Screenshot 2022-09-15 at 12 32 31 AM" src="https://user-images.githubusercontent.com/4344846/190316866-24aba2f8-aeae-40c6-83e3-ed29fab1ba46.png">
<img width="1176" alt="Screenshot 2022-09-15 at 12 32 17 AM" src="https://user-images.githubusercontent.com/4344846/190316878-aa9ea953-93e6-4885-a5da-eb272ef6bdda.png">

The job of Expression Profiler appears to be to reduce projection batch sizes for expensive expressions so that they don't hog CPU for too long. Any change to projection batch size due to this heuristic does not impact the size of reads and the pages produced by orc/parquet (those are based on different criteria like locality of reads and memory consumption). So it seems that Expression Profiler should ignore time taken to load page from the page source.